includes/class-newspack-newsletters.php

```diff
@@ -706,22 +707,40 @@ public static function api_set_mailchimp_interest( $request ) {
 				);
 			}
 
-			$segment_opts = ( 'no_interests' === $request['interest_id'] ) ?
-				(object) [] :
-				[
+			$has_interest  = $interest_id_param && ! $is_unsetting_interest;
+			$tag_ids_param = $request['tag_ids'];
+
+			$segment_opts = (object) [];
+
+			if ( $has_interest || $tag_ids_param ) {
+				$segment_opts = [
 					'match'      => 'any',
-					'conditions' => [
-						[
-							'condition_type' => 'Interests',
-							'field'          => $field,
-							'op'             => 'interestcontains',
-							'value'          => [
-								$interest_id,
-							],
-						],
-					],
+					'conditions' => [],
 				];
 
+				if ( $has_interest ) {
+					$segment_opts['conditions'][] = [
+						'condition_type' => 'Interests',
+						'field'          => $field,
+						'op'             => 'interestcontains',
+						'value'          => [
+							$interest_id,
+						],
+					];
+				}
+
+				if ( $tag_ids_param ) {
+					foreach ( $tag_ids_param as $tag_id ) {
+						$segment_opts['conditions'][] = [
+							'condition_type' => 'StaticSegment',
+							'field'          => 'static_segment',
+							'op'             => 'static_is',
+							'value'          => $tag_id,
+						];
+					}
+				}
+			}
+
```

Comment:
If you specify an Interest and Tags, Mailchimp will treat the segmentation as OR logic, which would mean if no subscribers match the tags the campaign will still be sent to every subscriber in the Interest. Here's what the segmentation logic in MC looks like for a campaign where I specified both:

<img width="1512" alt="Screen Shot 2020-07-05 at 2 44 33 PM" src="https://user-images.githubusercontent.com/1477002/86539917-73966d00-bece-11ea-86ea-37d41c2966df.png">

I'm not 100% sure, but I _think_ we would want this segment to be all subscribers in that Interest Group bearing the specified Tags?

Reply:
Treating this as AND logic makes sense, fixed in adb5a8031b587ed3db273aec70da5eb5ee336909